### PR TITLE
Provider 'could not detect network' backoff timer

### PR DESCRIPTION
**Pull request by jdowning100**
This PR also disables analytics.

### CODE REVIEW

File: background/services/analytics/index.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export default class AnalyticsService extends BaseService<Events> {
         hasDefaultOnBeenTurnedOn,
       })
 
-      await this.emitter.emit("enableDefaultOn", undefined)
+      //await this.emitter.emit("enableDefaultOn", undefined)
```
Comment (gameofpointers) [inline background/services/analytics/index.ts]: comment? or maybe remove the line, if we want analytics we can create an issue to add it

File: background/lib/posthog.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const POSTHOG_URL =
 export const USE_ANALYTICS_SOURCE = process.env.USE_ANALYTICS_SOURCE
 
 export function shouldSendPosthogEvents(): boolean {
-  return !!process.env.POSTHOG_API_KEY
+  return false //!!process.env.POSTHOG_API_KEY
```
Comment (gameofpointers) [inline background/lib/posthog.ts]: comment

File: background/lib/posthog.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ export const isOneTimeAnalyticsEvent = (
   return Object.values<string>(OneTimeAnalyticsEvent).includes(eventName)
 }
 
-const POSTHOG_PROJECT_ID = "11112"
+const POSTHOG_PROJECT_ID = "0"
```
Comment (gameofpointers) [inline background/lib/posthog.ts:29]: what is this ID?

File: background/services/chain/serial-fallback-provider.ts [hunk 4]
```diff
@@ -754,17 +762,39 @@ export default class SerialFallbackProvider extends QuaisJsonRpcProvider {
...
     logger.debug(
       "Reconnecting provider at index",
       this.currentProviderIndex,
       "..."
     )
-
-    this.currentProvider = this.providerCreators[this.currentProviderIndex].creator()
-    await this.resubscribe(this.currentProvider)
-
-    // TODO After a longer backoff, attempt to reset the current provider to 0.
```
Comment (gameofpointers) [inline background/services/chain/serial-fallback-provider.ts:767]: is the second part of this TODO valid now? i dont see the current provider set to 0

File: background/lib/posthog.ts (hunk 3 above)
Comment (jdowning100) [reply background/lib/posthog.ts:29]: It's a default ID, used in the posthog endpoint URL. Probably won't work, but Posthog isn't accessed with this commit anyways.

File: background/services/chain/serial-fallback-provider.ts (hunk 4 above)
Comment (jdowning100) [reply background/services/chain/serial-fallback-provider.ts:767]: This TODO is no longer valid

File: background/services/analytics/index.ts (hunk 1 above)
Comment (jdowning100) [reply background/services/analytics/index.ts]: Removed

File: background/lib/posthog.ts (hunk 2 above)
Comment (jdowning100) [reply background/lib/posthog.ts]: Removed

File: background/lib/posthog.ts (hunk 3 above)
Comment (wizeguyy) [reply background/lib/posthog.ts:29]: Does it need to change as part of this PR? I'd prefer to only change what needs to change

Comment (jdowning100) [reply background/lib/posthog.ts:29]: This PR also disables analytics. I updated the description to reflect that.

File: background/services/chain/serial-fallback-provider.ts [hunk 5]
```diff
@@ -754,17 +762,39 @@ export default class SerialFallbackProvider extends QuaisJsonRpcProvider {
...
+              "ms. URL:",
+              this.currentProvider.connection.url
+            )
+            this.readyForReconnect = false
+            setTimeout(() => this.readyForReconnect = true, this.backoffTime)
+            this.backoffTime = Math.min(this.backoffTime * 2, MAX_BACKOFF_MS)
+          }
+        }
+    } finally {
+      if(!err && this.backoffTime != BASE_BACKOFF_MS) {
```
Comment (wizeguyy) [inline background/services/chain/serial-fallback-provider.ts:791]: Do we need the second half of this conditional? Seems like, no matter what, `this.backoffTime` should be `BASE_BACKOFF_MS` if there wasn't an error.

**Review by wizeguyy — APPROVED** [review]

**Review by gameofpointers — APPROVED** [review]

File: background/services/chain/serial-fallback-provider.ts (hunk 5 above)
Comment (jdowning100) [reply background/services/chain/serial-fallback-provider.ts:791]: If there was an error and then after a subsequent call there isn't, this.backoffTime should be reset to BASE_BACKOFF_MS

**Pull request merged**